driver/src/main/java/org/neo4j/driver/internal/SimpleRecord.java

```diff
@@ -57,7 +58,16 @@ public Value get( int fieldIndex )
     @Override
     public Value get( String fieldName )
     {
-        return fields[fieldLookup.get( fieldName )];
+        Integer fieldIndex = fieldLookup.get( fieldName );
+
+        if ( fieldIndex == null )
+        {
+            throw new ValueNotFoundException( fieldName );
```

Comment:
The strategy to guide the changes made in this PR is to make it easier for a end user to find their error. So I feel a client exception at the Record interface is easier than a NPE for a user to find the error. But if the cypher behaviour is already defined to return null, we definitely have no reason to change that. 

I will update the pr to change to return null and add doc instead.